### DISCUSSION TITLE
How to use Image column with a URL

### DISCUSSION THREAD

**Discussion by notsidney** [AntlerVC/firetable#724] [Support / Q&A] [answered]
I store only the URLs of images. How can I display them in an Image column?

![Screenshot of table with URL column and Derivative column that renders an image](https://user-images.githubusercontent.com/27017118/180116938-37d11646-348b-49be-b67c-f1aa7fe1a7ce.png)

**Comment by notsidney** [comment, marked as answer]
We don’t currently support displaying just URLs directly in the Image column itself. We’re looking into how we can support other formats for storing images.

### Temporary workaround: use Derivative columns

However, you can create a [Derivative column](https://docs.rowy.io/field-types/derivative) (if you have Rowy Run installed). Derivative columns display a value derived from another column’s value.

When you create a Derivative column, change the config as follows:
<img width="928" alt="image" src="https://user-images.githubusercontent.com/27017118/180120568-11df98ac-50a0-4b41-acc9-221a3c0e17cb.png">

1. Set the **Listener fields** to be the URL column.
2. Change the **Output field type** to Image.
3. Set the **Derivative script** to this:
   ```ts
   const derivative: Derivative = async ({ row, ref, db, storage, auth }) => {
     return [{ downloadURL: row.URL_FIELD_KEY }]
   }
   ```
   Make sure to change **URL_FIELD_KEY** to the **field key** of the URL column.
4. **Lock** the Derivative column to prevent people uploading to this column directly.
5. Continue to add images by providing the URL in the URL column.

**Reply by KKimj** [reply]
Thanks!! 
It helped a lot.
Have a Nice Day!
